app/support/products/scheduling_support.rb

```diff
@@ -155,14 +155,19 @@ class ReservationFinder
     attr_accessor :time, :rule, :day_start, :day_end, :options
 
     def initialize(time, rule, options = {})
-      self.time = time
+      self.time = rule.instrument.cutoff_hours > 0 ? time_with_cutoff_hours(time, rule.instrument) : time
       self.rule = rule
       self.day_start = Time.zone.local(time.year, time.month, time.day, rule.start_hour, rule.start_min, 0)
       self.day_end   = Time.zone.local(time.year, time.month, time.day, rule.end_hour, rule.end_min, 0)
       self.options   = options
       adjust_time
     end
 
+    def time_with_cutoff_hours(time, instrument)
+      first_available_time = Time.zone.now + instrument.cutoff_hours.hours
```

Comment:
I think we like to prefer `hours.from_now`. And in general, we're preferring `Time.current` to `Time.zone.now`, but that's not a big deal.